resolver/src/lib.rs

```diff
@@ -0,0 +1,42 @@
+//! TODO
+
+use std::future::Future;
+
+use bytes::Bytes;
+use commonware_utils::Serialize;
+use futures::channel::oneshot;
+use std::fmt::Debug;
+
+pub mod p2p;
+
+/// Used as the key to uniquely identify data.
+pub trait Key: Serialize + Clone + Debug + PartialEq + Eq + Ord + Send + Sync + 'static {}
+
+/// Client is the interface responsible for fetching data from the network.
+pub trait Client: Clone + Send + 'static {
```

Comment:
As discussed, let's add an option for the client to `drop` a request by `Key` (in case this is used in some ordered context where we may no longer care about some dependency at some point/prevents an infinite loop fetching something that can't be fetched).